src/autoupdate.c

```diff
@@ -89,54 +109,71 @@ static void policy_warn(void)
 
 enum swupd_code autoupdate_main(int argc, char **argv)
 {
+
 	if (!parse_options(argc, argv)) {
+		print_help();
 		return SWUPD_INVALID_OPTION;
 	}
-	if (enable && disable) {
-		error("Can not enable and disable at the same time\n");
-		return SWUPD_INVALID_OPTION;
-	}
-	if (!systemctl_active()) {
+
+	if (!globals.path_prefix && !systemctl_active()) {
 		error("Systemd is inactive - unable to proceed\n");
 		return SWUPD_SUBPROCESS_ERROR;
 	}
 
-	if (enable) {
+	if (cmdline_option_enable) {
 		int rc;
 		check_root();
 		info("Running systemctl to enable updates\n");
-		rc = systemctl_cmd("unmask", "--now", "swupd-update.service", "swupd-update.timer", NULL);
+		rc = systemctl_cmd_path(globals.path_prefix,
+					"unmask", "--now", "swupd-update.service", "swupd-update.timer", NULL);
 		if (rc != 0) {
 			return SWUPD_SUBPROCESS_ERROR;
 		}
-		rc = systemctl_restart("swupd-update.timer");
+
+		/* If globals.path_prefix is given and we have succeeded so far,
+		 * make sense to return SWUPD_OK at this point. In that case, 
+		 * a swupd timer restart cannot be done as there is no
+		 * system daemon to communicate for rootfs.
+		 */
+		if (globals.path_prefix) {
+			warn("Running autoupdate with --path will not restart swupd-update.timer. This will have to be done manually\n");
+			return SWUPD_OK;
+		}
+
+		rc = systemctl_cmd_path(globals.path_prefix, "restart", "swupd-update.timer", NULL);
 		if (rc != 0) {
 			return SWUPD_SUBPROCESS_ERROR;
 		}
 
 		return SWUPD_OK;
-	} else if (disable) {
+	} else if (cmdline_option_disable) {
 		int rc;
 		check_root();
 		policy_warn();
 		info("Running systemctl to disable updates\n");
-		rc = systemctl_cmd("mask", "--now", "swupd-update.service", "swupd-update.timer", NULL);
+		rc = systemctl_cmd_path(globals.path_prefix,
+					"mask", "--now", "swupd-update.service", "swupd-update.timer", NULL);
 		if (rc) {
 			return SWUPD_SUBPROCESS_ERROR;
 		}
 		return SWUPD_OK;
 	} else {
 		int rc1, rc2;
 		const int STATUS_UNKNOWN = 4;
-		rc1 = systemctl_cmd("is-enabled", "swupd-update.service", NULL);
-		rc2 = systemctl_cmd("is-active", "swupd-update.timer", NULL);
+		rc1 = systemctl_cmd_path(globals.path_prefix, "is-enabled", "swupd-update.service", NULL);
+		rc2 = systemctl_cmd_path(globals.path_prefix, "is-active", "swupd-update.timer", NULL);
 		if (rc1 == SWUPD_OK && rc2 == SWUPD_OK) {
 			print("Enabled\n");
 			return SWUPD_OK;
 		} else if (rc1 >= STATUS_UNKNOWN || rc2 >= STATUS_UNKNOWN) {
 			/* systemctl returns 1,2, or 3 when program dead or not running */
 			error("Unable to determine autoupdate status\n");
 			return SWUPD_SUBPROCESS_ERROR;
+			// Swupd-service is unmasked, static but timer is inactive for --path
+		} else if (rc1 == SWUPD_OK && rc2 == (STATUS_UNKNOWN - 1) && globals.path_prefix) {
```

Comment:
Magic number STATUS_UNKNOWN -1. Is this behavior defined in systemctl documentation?

Reply:
My suggestion on that is just don't run is-active when path_prefix != "/"

Reply:
I can do that..problem is the globals.path_prefix has no default value of '/' value in it as we dont perform SWUPD_INIT ad SWUPD_DEINIT in case of autoupdate as it is not really required. Should we?

Reply:
You rightly pointed out ..there is no documentation for isactive for Magic number STATUS_UNKNOWN -1. The only thing they say it returns a number other than zero for in-active.

Reply:
Done